crates/task-impls/src/rewind.rs

```diff
@@ -0,0 +1,69 @@
+use std::sync::Arc;
+
+use anyhow::Result;
+use async_broadcast::{Receiver, Sender};
+use async_trait::async_trait;
+use hotshot_task::task::TaskState;
+use hotshot_types::traits::node_implementation::NodeType;
+use std::fs::OpenOptions;
+use std::io::Write;
+
+use crate::events::HotShotEvent;
+
+/// The task state for the `Rewind` task is used to capture all events received
+/// by a particular node, in the order they've been received.
+pub struct RewindTaskState<TYPES: NodeType> {
+    /// All events received by this node since the beginning of time.
+    pub events: Vec<Arc<HotShotEvent<TYPES>>>,
```

Comment:
This is going to be a big memory leak if we run this on staging or similar.  Maybe the task can checkpoint every 100-1000 views to clear the vector and realease the `HotShotEvent` (I assume this task will be the only task holding the `Arc` by then

Reply:
If it's just for integration tests/demo/benchmark runs then feel free to ignore

Reply:
It *is* right now, but I'd like to have this be used in deployments as well. I had mentioned in the comment above to add a sliding window for production use (right now it won't let you deploy with a release configuration at all), but I'd be open to discussing an implementation now if you think it's worth tossing in.

Reply:
I think we can just check event view numbers and every time they are over a certain threshold we flush to a file.  Since this task isn't really on a critical path it can be as slow as we want.  So basically set a value called `threshold_view` and check some of the events (maybe `QuorumProposalValidated`) view number to be creater than theshold.  If it is set `threshould_view` to `threshold_view + WINDOW_SIZE` and spawn a task to write to a file/db or whatever.

We could also layer on top of that a sliding window where we just write to 10 different files and overwrite the first on the 11th window.  

Basically I don't think it needs to be anything fancy and it can be addressed in another PR